src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,20 +99,64 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     float professorScore = professorSentiment.getScore();
     profLanguageService.close();
 
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
-
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    // Gets user email.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
 
+    // Check whether user has reviewer ID in system.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    Filter userFilter = new FilterPredicate("user-email", FilterOperator.EQUAL, userEmail);
+    Query userQuery = new Query("User").setFilter(userFilter);
+    List<Entity> userQueryList =
+        datastore.prepare(userQuery).asList(FetchOptions.Builder.withDefaults());
+    Entity userEntity = new Entity("User");
 
+    // User has not reviewed any rating.
+    if (userQueryList.isEmpty()) {
+      Entity newReviewer = new Entity("User");
+      newReviewer.setProperty("user-email", userEmail);
+      userEntity = newReviewer;
+    } else {
+      userEntity = userQueryList.get(0);
+    }
+
+    // Has to be added to URL.
+    currentTermKey = request.getParameter("term");
+    Entity currentTerm = datastore.get(currentTermKey);
+
+    // Checks for term rating by reviewer.
+    Filter termReviewerFilter =
+        new FilterPredicate("reviewer-id", FilterOperator.EQUAL, userEntity.getKey());
+    Query termRatingQuery =
+        new Query("Rating Term").setAncestor(currentTermKey).setFilter(termReviewerFilter);
+    List<Entity> termRatingQueryList =
+        datastore.prepare(termRatingQuery).asList(FetchOptions.Builder.withDefaults());
+
+    if (termRatingQueryList.isEmpty()) {
```

Comment:
This if-else has a lot of duplicated code, it looks like you can DRY it up

Reply:
Ah, nice and DRY : ) You could shorten this even more if you wanted with a ternary:
```Entity classRatingEntity = termRatingQueryList.isEmpty() ? new Entity("Rating Term", currentTermKey) : termRatingQueryList.get(0);```

Up to you if which way you like best for readability